### PR TITLE
[Merged by Bors] - chore: bump Std (adapt to migration of ListM)

### PR DESCRIPTION
**Pull request by kim-em**
`ListM` has been [moved to `Std`](https://github.com/leanprover/std4/pull/165) and renamed as `MLList`.

There are some slight API functions, that ensure operations are lazy, requiring the addition of some `fun _ =>` here in Mathlib.

---

- [x] depends on: https://github.com/leanprover/std4/pull/165
- [x] depends on: #6435

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by eric-wieser — APPROVED** [review]
Can you summarize the changes in the PR description?

bors d+

**Comment by semorrison** [comment]
bors merge
